### PR TITLE
[SD-663] Corrects breadcrumbs in frontend_helper.rb

### PR DESCRIPTION
**Pull request by PSzyszkaSpark**
Fixes https://spark-solutions.atlassian.net/browse/SD-663

### CODE REVIEW

File: frontend/app/helpers/spree/frontend_helper.rb [hunk 1]
```diff
@@ -11,11 +11,11 @@ def spree_breadcrumbs(taxon, separator = '', product = nil)
...
+        crumbs << taxon.ancestors.where.not(parent_id: nil).each_with_index.map { |ancestor, index| content_tag(:li, content_tag(:span, link_to(content_tag(:span, ancestor.name, itemprop: 'name'), seo_url(ancestor, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', itemid: seo_url(ancestor, params: permitted_product_params), class: 'breadcrumb-item', position: index + 1) } if taxon.ancestors.any?
+        crumbs << content_tag(:li, content_tag(:span, link_to(content_tag(:span, taxon.name, itemprop: 'name'), seo_url(taxon, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), class: 'breadcrumb-item', itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', itemid: seo_url(taxon, params: permitted_product_params), position: taxon.ancestors.where.not(parent_id: nil).count + 1)
+        crumbs << content_tag(:li, content_tag(:span, content_tag(:span, product.name) + separator), class: 'breadcrumb-item', itemid: spree.product_path(product, taxon_id: taxon.try(:id)), position: 0) if product
```
Comment (damianlegawiec) [inline frontend/app/helpers/spree/frontend_helper.rb]: @PSzyszkaSpark shouldn't this  be a subsequent number, eg. if we have 2 taxons it would be 3 ?

**Review by pmilewski — APPROVED** [review]

File: frontend/app/helpers/spree/frontend_helper.rb [hunk 2]
```diff
@@ -11,11 +11,11 @@ def spree_breadcrumbs(taxon, separator = '', product = nil)
...
+        crumbs << content_tag(:li, content_tag(:span, link_to(content_tag(:span, taxon.name, itemprop: 'name'), seo_url(taxon, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), class: 'breadcrumb-item', itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', itemid: seo_url(taxon, params: permitted_product_params), position: taxon.ancestors.where.not(parent_id: nil).count + 1)
+        crumbs << content_tag(:li, content_tag(:span, content_tag(:span, product.name) + separator), class: 'breadcrumb-item', itemid: spree.product_path(product, taxon_id: taxon.try(:id)), position: taxon.ancestors.where.not(parent_id: nil).count + 2) if product
```
Comment (damianlegawiec) [inline frontend/app/helpers/spree/frontend_helper.rb]:
```suggestion
        ancestors =  taxon.ancestors.where.not(parent_id: nil)
        crumbs << ancestors.each_with_index.map { |ancestor, index| content_tag(:li, content_tag(:span, link_to(content_tag(:span, ancestor.name, itemprop: 'name'), seo_url(ancestor, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', itemid: seo_url(ancestor, params: permitted_product_params), class: 'breadcrumb-item', position: index + 1) }
        crumbs << content_tag(:li, content_tag(:span, link_to(content_tag(:span, taxon.name, itemprop: 'name'), seo_url(taxon, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), class: 'breadcrumb-item', itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', itemid: seo_url(taxon, params: permitted_product_params), position: ancestors.size + 1
        content_tag(:li, content_tag(:span, content_tag(:span, product.name) + separator), class: 'breadcrumb-item', itemid: spree.product_path(product, taxon_id: taxon.try(:id)), position: ancestors.size + 2) if product
```

This way we're reducing number of SQL queries and keeping everything in memory making it more performant

**Review by damianlegawiec — CHANGES_REQUESTED** [review]

File: frontend/app/helpers/spree/frontend_helper.rb [hunk 3]
```diff
@@ -11,11 +11,12 @@ def spree_breadcrumbs(taxon, separator = '', product = nil)
...
+        crumbs << content_tag(:li, content_tag(:span, link_to(content_tag(:span, taxon.name, itemprop: 'name'), seo_url(taxon, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), class: 'breadcrumb-item', itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', itemid: seo_url(taxon, params: permitted_product_params), position: ancestors.size + 1
+        content_tag(:li, content_tag(:span, content_tag(:span, product.name) + separator), class: 'breadcrumb-item', itemid: spree.product_path(product, taxon_id: taxon.try(:id)), position: ancestors.size + 2) if product
       else
         crumbs << content_tag(:li, content_tag(:span, Spree.t(:products), itemprop: 'item'), class: 'active', itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement')
```
Comment (damianlegawiec) [inline frontend/app/helpers/spree/frontend_helper.rb]:
```suggestion
        crumbs << content_tag(:li, content_tag(:span, Spree.t(:products), itemprop: 'item'), class: 'active', itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', position: 1)
```

File: frontend/app/helpers/spree/frontend_helper.rb [hunk 4]
```diff
@@ -11,11 +11,12 @@ def spree_breadcrumbs(taxon, separator = '', product = nil)
...
       if taxon
-        crumbs << taxon.ancestors.where.not(parent_id: nil).map { |ancestor| content_tag(:li, content_tag(:span, link_to(content_tag(:span, ancestor.name, itemprop: 'name'), seo_url(ancestor, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement', class: 'breadcrumb-item') } unless taxon.ancestors.empty?
-        crumbs << content_tag(:li, content_tag(:span, link_to(content_tag(:span, taxon.name, itemprop: 'name'), seo_url(taxon, params: permitted_product_params), itemprop: 'url') + separator, itemprop: 'item'), class: 'breadcrumb-item', itemscope: 'itemscope', itemtype: 'https://schema.org/ListItem', itemprop: 'itemListElement')
-        crumbs << content_tag(:li, content_tag(:span, content_tag(:span, product.name) + separator), class: 'breadcrumb-item') if product
+        ancestors =  taxon.ancestors.where.not(parent_id: nil)
```
Comment (damianlegawiec) [inline frontend/app/helpers/spree/frontend_helper.rb]:
```suggestion
        ancestors = taxon.ancestors.where.not(parent_id: nil)
```

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**